### PR TITLE
create file format tag --fileformat for converting audio downloads 

### PR DESCRIPTION
**Pull request by caitriggs**
Addresses feature request #6

### CODE REVIEW

**Comment by carlthome** [comment]
Great job! Could you add a test module (e.g. https://github.com/caitriggs/audio-scraper/blob/fileformat/tests/test_audioconvert.py) of the builtin audio conversion functionality to make sure it works?

File: audioscrape/__main__.py [hunk 1]
```diff
@@ -2,14 +2,20 @@
...
 from . import soundcloud, youtube
 
 
-def download(query, include=None, exclude=None, quiet=False, overwrite=False):
+def download(query, include=None, exclude=None, quiet=False,
+             overwrite=False, fileformat=None):
     """Scrape various websites for audio."""
-    youtube.scrape(query, include, exclude, quiet, overwrite)
-    soundcloud.scrape(query, include, exclude, quiet, overwrite)
+    # create subdirectory for converted audio files if --fileformat tag set
```
Comment (carlthome) [inline audioscrape/__main__.py]: Minor nitpick but capitalize and have normal sentences with periods for code comments, please. 🐼

File: audioscrape/audioconvert.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import subprocess
+
+
+def ffmpeg_convert(file, audio_title, fileformat):
+    command = ['ffmpeg', '-hide_banner',  # quiet ffmpeg banner
+               '-loglevel', 'panic',   # quiet ffmpeg stdout
+               '-i', "./{}".format(file),  # input file to convert
+               '-f', '{}'.format(fileformat),  # output fileformat type
+               '-ac', '1',  # mono channel
+               '-ar', '16000',  # sampling rate 16000Hz
```
Comment (carlthome) [inline audioscrape/audioconvert.py]: Should 16 kHz mono be the default or is 44100 Hz / 16-bit stereo more common? What are the intended use cases of the audio conversion?

File: audioscrape/soundcloud.py [hunk 3]
```diff
@@ -95,3 +96,12 @@ def pagination(x):
                             unit='MB',
                             file=sys.stdout):
                         f.write(data)
+                '''
+                Appears that soundcloud streams are mp3 only, so we'll
+                convert original mp3 to user defined file format with ffmpeg.
+                '''
+                # Convert to fileformat using ffmpeg
```
Comment (carlthome) [inline audioscrape/soundcloud.py]: Condense the two comments into one.

File: audioscrape/youtube.py [hunk 4]
```diff
@@ -51,3 +52,16 @@ def scrape(query, include, exclude, quiet, overwrite):
 
         # Download audio to working directory.
         audio.download(quiet=quiet)
+
+        '''
+        Since pafy.Stream object (audio) does not appear to grab audio content
+        itself until Stream.download(), we must convert
+        the audio after download with ffmpeg.
+        '''
+        # Convert to fileformat using ffmpeg
```
Comment (carlthome) [inline audioscrape/youtube.py]: Condense the two comments into one.

**Review by carlthome — CHANGES_REQUESTED** [review]
Great job!

I want to keep this package as lightweight as possible so depending on ffmpeg is a little scary but I also see the benefit of getting mp3:s, Ogg:s or whatever out instead of a mix of formats.

Could you add a brief test of the audio conversion module (e.g https://github.com/caitriggs/audio-scraper/blob/fileformat/tests/test_audioconvert.py) so we can make sure it works?

File: audioscrape/audioconvert.py (hunk 2 above)
Comment (caitriggs) [reply audioscrape/audioconvert.py]: 44100 Hz stereo is better quality, but the specific application I'm using your scraper for is for ASR model training which are generally built for 16000 Hz mono channel wav files (e.g. Bai Du's DeepSpeech). I should probably re-tool it to allow user defined settings, but admit I just added a quick and dirty hard coded solution for my needs

File: audioscrape/__main__.py (hunk 1 above)
Comment (caitriggs) [reply audioscrape/__main__.py]: This is your show, so no problem there! Will do 👍

**Comment by caitriggs** [comment]
> I want to keep this package as lightweight as possible so depending on ffmpeg is a little scary but I also see the benefit of getting mp3:s, Ogg:s or whatever out instead of a mix of formats.

Yea, I totally understand keeping the dependency footprint small, but for me it's really convenient to build file conversion into the scraping process. Many of us in the data science community scrape data for a specific purpose (e.g. all input files have to be 16kHz mono wave files). 

ffmpeg is a pretty well established standard for audio/video encoding/decoding so thought it was safe library choice. Especially since if you're working with audio files in the first place, you're already likely to have ffmpeg installed.

> Could you add a brief test of the audio conversion module (e.g https://github.com/caitriggs/audio-scraper/blob/fileformat/tests/test_audioconvert.py) so we can make sure it works?

Yup, shouldn't be an issue 👍

**Comment by carlthome** [comment]
Closing because stale.

**Pull request closed without merging**